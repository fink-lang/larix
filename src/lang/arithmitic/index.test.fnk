{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse_expr, parse} = import '../../'


describe 'arithmitic', fn:

  it 'parses +', fn:
    expect
      parse_expr `a + b`
      to_match_snapshot

    expect
      parse_expr `a + b * c`
      to_match_snapshot


  it 'parses * ', fn:
    expect
      parse_expr `a * b`
      to_match_snapshot

    expect
      parse_expr `a * b * c`
      to_match_snapshot

    expect
      parse_expr `a * b + c`
      to_match_snapshot

  it 'parses / ', fn:
    expect
      parse_expr `a / b`
      to_match_snapshot

    expect
      parse_expr `a / b / c`
      to_match_snapshot

    expect
      parse_expr `a / b + c`
      to_match_snapshot


  it 'parses %', fn:
    expect
      parse_expr `a % b`
      to_match_snapshot

    expect
      parse_expr `a % b % c`
      to_match_snapshot

    expect
      parse_expr `a % b / c`
      to_match_snapshot


  it 'parses ^', fn:
    expect
      parse_expr `a ^ b`
      to_match_snapshot

    expect
      parse_expr `a ^ b ^ c`
      to_match_snapshot

    expect
      parse_expr `a ^ b * c`
      to_match_snapshot

    expect
      parse_expr `a * b ^ c`
      to_match_snapshot


  it 'parses - prefix', fn:
    expect
      parse_expr `-a`
      to_match_snapshot

    expect
      parse_expr `-a - b`
      to_match_snapshot

    expect
      parse_expr `-a * b`
      to_match_snapshot

    expect
      parse_expr `a * -b`
      to_match_snapshot

    expect
      parse_expr `foo -b`
      to_match_snapshot

    expect
      parse `
        foo bar
        -spam
        `
      to_match_snapshot


  it 'parses -', fn:
    expect
      parse_expr `a - b`
      to_match_snapshot

    expect
      parse_expr `a - b - c`
      to_match_snapshot

    expect
      parse_expr `a - b + c`
      to_match_snapshot


  it 'parses grouped', fn:
    expect
      parse_expr `(a - b) * c`
      to_match_snapshot


  it 'parses multiline', fn:
    expect
      parse_expr `
        a
          + b +
          c
            + d
      `
      to_match_snapshot



