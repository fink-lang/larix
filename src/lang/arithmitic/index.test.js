import {parse_expr} from '../../';


describe('arithmitic', ()=> {

  it('parses single line: a + b * c', ()=> {
    expect(
      parse_expr(`a + b * c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a * b + c', ()=> {
    expect(
      parse_expr(`a * b + c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a % b / c', ()=> {
    expect(
      parse_expr(`a % b / c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a ^ b * c', ()=> {
    expect(
      parse_expr(`a ^ b * c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a * -b', ()=> {
    expect(
      parse_expr(`a * -b`)
    ).toMatchSnapshot();
  });


  it('parses grouped: (a - b) * c', ()=> {
    expect(
      parse_expr(`(a - b) * c`)
    ).toMatchSnapshot();
  });
});
