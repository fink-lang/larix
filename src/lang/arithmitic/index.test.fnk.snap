// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arithmitic parses % 1`] = `
arithm % (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses % 2`] = `
arithm % (1:0-1:9)
  arithm % (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses % 3`] = `
arithm / (1:0-1:9)
  arithm % (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses *  1`] = `
arithm * (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses *  2`] = `
arithm * (1:0-1:9)
  arithm * (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses *  3`] = `
arithm + (1:0-1:9)
  arithm * (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses + 1`] = `
arithm + (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses + 2`] = `
arithm + (1:0-1:9)
  ident (1:0-1:1) a
  arithm * (1:4-1:9)
    ident (1:4-1:5) b
    ident (1:8-1:9) c
`;

exports[`arithmitic parses /  1`] = `
arithm / (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses /  2`] = `
arithm / (1:0-1:9)
  arithm / (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses /  3`] = `
arithm + (1:0-1:9)
  arithm / (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses ^ 1`] = `
arithm:right ^ (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses ^ 2`] = `
arithm:right ^ (1:0-1:9)
  ident (1:0-1:1) a
  arithm:right ^ (1:4-1:9)
    ident (1:4-1:5) b
    ident (1:8-1:9) c
`;

exports[`arithmitic parses ^ 3`] = `
arithm * (1:0-1:9)
  arithm:right ^ (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses ^ 4`] = `
arithm * (1:0-1:9)
  ident (1:0-1:1) a
  arithm:right ^ (1:4-1:9)
    ident (1:4-1:5) b
    ident (1:8-1:9) c
`;

exports[`arithmitic parses - 1`] = `
arithm - (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`arithmitic parses - 2`] = `
arithm - (1:0-1:9)
  arithm - (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses - 3`] = `
arithm + (1:0-1:9)
  arithm - (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`arithmitic parses - prefix 1`] = `
arithm:prefix - (1:0-1:2)
  ident (1:1-1:2) a
`;

exports[`arithmitic parses - prefix 2`] = `
arithm - (1:0-1:6)
  arithm:prefix - (1:0-1:2)
    ident (1:1-1:2) a
  ident (1:5-1:6) b
`;

exports[`arithmitic parses - prefix 3`] = `
arithm:prefix - (1:0-1:6)
  arithm * (1:1-1:6)
    ident (1:1-1:2) a
    ident (1:5-1:6) b
`;

exports[`arithmitic parses - prefix 4`] = `
arithm * (1:0-1:6)
  ident (1:0-1:1) a
  arithm:prefix - (1:4-1:6)
    ident (1:5-1:6) b
`;

exports[`arithmitic parses - prefix 5`] = `
call (1:0-1:6)
  ident (1:0-1:3) foo
  arithm:prefix - (1:4-1:6)
    ident (1:5-1:6) b
`;

exports[`arithmitic parses - prefix 6`] = `
module (1:0-3:0)
  call (1:0-1:7)
    ident (1:0-1:3) foo
    ident (1:4-1:7) bar
  arithm:prefix - (2:0-2:5)
    ident (2:1-2:5) spam
`;

exports[`arithmitic parses grouped 1`] = `
arithm * (1:0-1:11)
  group (1:0-1:7)
    arithm - (1:1-1:6)
      ident (1:1-1:2) a
      ident (1:5-1:6) b
  ident (1:10-1:11) c
`;

exports[`arithmitic parses multiline 1`] = `
arithm + (1:0-4:7)
  arithm + (1:0-3:3)
    arithm + (1:0-2:5)
      ident (1:0-1:1) a
      ident (2:4-2:5) b
    ident (3:2-3:3) c
  ident (4:6-4:7) d
`;
